src/gerbil/compiler/driver.ss

```diff
@@ -200,7 +200,9 @@ namespace: gxc
              (else
               (raise-compile-error "libgerbil does not exist" libgerbil.a libgerbil.so))))
            (gerbil-rpath
-            (string-append "-Wl,-rpath=" gerbil-libdir))
+            (string-append
+	     (cond-expand (darwin "-Wl,-rpath,") (else "-Wl,-rpath="))
+	     gerbil-libdir))
```

Comment:
you need to do it in `compile-exe/full-program-optimization` as well, let's make it a top level procedure named `(gerbil-rpath)`.